Add code for motor controllers

ryanloeffelman opened the pull request:
Added code to run on the arduinos controlling the motors
Closes #6

Commit:
Add code for motor controllers

Added code to run on the arduinos controlling the motors

matthewia94 commented:
@ryanloeffelman Please address the issues I have marked.

matthewia94 commented on ros_ws/src/arudino/motor/motor.ino:
```diff
@@ -0,0 +1,84 @@
+
+//ROS Subscriber for the Velocity Message
+ros::Subscriber<std_msgs::Float64> 
+     velocitySub("wheel_velocity", &VelocityCallback);
```
@islamelnabarawy Any suggestions on how we should handle constants like this in the arduino files?  Including the common/constants.h becomes a mess when working with the arduino IDE as far as I can tell you have to include it in your sketch or in your arduino libraries folder.

Commit:
Fix all issues except the include

Fixed all the issues you came up with however code will not run due to
needing to include a file that it can't find.

matthewia94 commented on ros_ws/src/arudino/motor/motor.ino:
```diff
@@ -0,0 +1,104 @@
+/**
+ * @brief ROS Subscriber for the Velocity Message
+ */
+ros::Subscriber<std_msgs::Float64> velocitySub(VELOCITY_TOPIC, &VelocityCallback);
```
Should be velocityCallback

matthewia94 commented on ros_ws/src/arudino/motor/motor.ino:
```diff
@@ -0,0 +1,104 @@
+const float MAX_RAD_SEC = (8*PI_CONST);
+
+/**
+ * @brief Pins used to controll the Motor
```
*control

matthewia94 commented on ros_ws/src/arudino/motor/motor.ino:
```diff
@@ -0,0 +1,104 @@
+const float PI_CONST = 3.14159265359;
+ 
+/**
+ * @brief Radians Per Second at full Throtall
```
*throtle

Commit:
Change Constant to a literal

Fixed spelling mistakes, and Changed the Constant "Velocity Topic" back to
"wheel_velocity"

Commit:
Add the ablity to scale the input as a float

Changed the way the input is scaled, from an integer value to accept
a floating point value.
Also allows the input to be scaled using a non-linear curve.

matthewia94 commented:
@ryanloeffelman Has this been tested?

Commit:
Remove all unecesarry build files

Removed all the Build Files so as to not clutter the workspace.
Also Added #define USB_CON to the arduino File

Commit:
Remove unnecessary constant

Remvoed an unnecessary constant form the Constants.h file

matthewia94 commented:
We haven't scaled the values yet.  This should be its own issue as it will be propagated through the non arduino code as well.
